pkg/plugin/validator.go

```diff
@@ -72,7 +73,10 @@ func (v *Validator) MatchIssue(ctx context.Context, issueURL string) error {
 	}
 	v.client = v.client.WithAuthToken(t)
 
-	return v.validateIssue(ctx, info)
+	if err := v.validateIssue(ctx, info); err != nil {
+		return errors.Join(errInvalidJustification, fmt.Errorf("failed to validated issue: %w", err))
```

Comment:
This also includes errors such as when the github client is down?

Reply:
Updated to only treat 404 as invalid justification error, also updated the code.

Reply:
maybe not just 404, see example: https://github.com/abcxyz/jvs-plugin-jira/blob/14a5b2698dd7eff63c9e786beb1ae619d9f17e29/pkg/plugin/validator.go#L201

Reply:
See this: https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#get-an-issue--status-codes

The rest api can only return these status code, and if an issue does not exist, it will return 404.